src/microbit/model/display.py

```diff
@@ -0,0 +1,207 @@
+                self.__image.blit(
+                    appended_image, x, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
+                )
+                time.sleep(delay / 1000)
+            if not loop:
+                break
+
+    def show(self, value, delay=400, wait=True, loop=False, clear=False):
+        if not wait:
+            thread = threading.Thread(
+                target=self.show, args=(value, delay, True, loop, clear)
+            )
+            thread.start()
+            return
+        while True:
+            if isinstance(value, Image):
+                self.__image = value.crop(
+                    0, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
+                )
+            elif isinstance(value, (str, int, float)):
+                if isinstance(value, str):
+                    chars = list(value)
+                else:
+                    chars = list(str(value))
+
+                for c in chars:
+                    self.__image = Display.__get_image_from_char(c)
+                    time.sleep(delay / 1000)
+            else:
+                # Check if iterable
+                try:
+                    _ = iter(value)
+                except TypeError as e:
+                    raise e
+
+                for elem in value:
+                    if isinstance(elem, Image):
+                        self.__image = elem.crop(
+                            0, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
+                        )
+                    elif isinstance(elem, str) and len(elem) == 1:
+                        self.__image = Display.__get_image_from_char(elem)
+                    # If elem is not char or image, break without iterating through rest of list
+                    else:
+                        break
+                    time.sleep(delay / 1000)
+            if not loop:
+                break
+        if clear:
+            self.clear()
+
+    def get_pixel(self, x, y):
+        return self.__image.get_pixel(x, y)
+
+    def set_pixel(self, x, y, value):
+        self.__image.set_pixel(x, y, value)
+
+    def clear(self):
+        self.__image = Image()
+
+    def on(self):
+        self.__on = True
+
+    def off(self):
+        self.__on = False
+
+    def is_on(self):
+        return self.__on
+
+    def read_light_level(self):
+        raise NotImplementedError(CONSTANTS.NOT_IMPLEMENTED_ERROR)
+
+    # Helpers
+
+    def __get_array(self):
+        return self.__image._Image__LED
+
+    def __print(self):
+        print("")
+        for i in range(CONSTANTS.LED_HEIGHT):
+            print(self._Display__image._Image__LED[i])
+
+    @staticmethod
+    def __get_image_from_char(c):
+        # If c is not between the ASCII alphabet we cover, make it a question mark
+        if ord(c) < CONSTANTS.ASCII_START or ord(c) > CONSTANTS.ASCII_END:
+            c = "?"
+        offset = (ord(c) - CONSTANTS.ASCII_START) * 5
+        representative_bytes = CONSTANTS.ALPHABET[offset : offset + 5]

```

Comment:
add a small comment to explain this? (particularly, why 5?)

Reply:
i updated it in the next commit, let me know what you think